Alvaro Gaming/PythonGaming_p3.py

```diff
@@ -14,79 +14,114 @@
 BLUE = (0, 0, 255)  # Blue
 GREEN = (0, 255, 0)  # Green
 
-
 # assigning FPS Value
 FPS = 60
 FramePerSec = pygame.time.Clock()
 
 # Setting up other variables that will be used in program
-SCREEN_WIDTH = 400
+SCREEN_WIDTH = 800
 SCREEN_HEIGHT = 600
 SPEED = 5
-SCORE = 0
+SCORE_1 = 0
+SCORE_2 = 0
 
-#Setting up Fonts
-font = pygame.font.SysFont("Verdana",60)
-font_small = pygame.font.SysFont("Verdana",20)
+# Setting up Fonts
+font = pygame.font.SysFont("Verdana", 60)
+font_small = pygame.font.SysFont("Verdana", 20)
 game_over = font.render("Game_Over", True, BLACK)
 
-background = pygame.image.load("AnimatedStreet.png")
+background = pygame.image.load("space.png")
 
 # Create white screen
 DISPLAYSURF = pygame.display.set_mode((SCREEN_WIDTH, SCREEN_HEIGHT))
 DISPLAYSURF.fill(WHITE)
-pygame.display.set_caption("Alvy Game")
+pygame.display.set_caption("Alvy Game: Pong")
+
 
 class Enemy(pygame.sprite.Sprite):
     def __init__(self):
         super().__init__()
         self.image = pygame.image.load("Enemy.png")
         self.rect = self.image.get_rect()
-        self.rect.center = (random.randint(40, SCREEN_WIDTH-40), 0)
+        #self.rect.center = (random.randint(40, SCREEN_WIDTH-40), 0)
+
+        self.velocity = [random.randint(4, 8), random.randint(-8, 8)]
 
+    """""
     def move(self):
-        self.rect.move_ip(0,10)
+        self.rect.move_ip(0, 10)
         if self.rect.bottom > 600:
             global SCORE
             SCORE += 1
             self.rect.top = 0
             self.rect.center = (random.randint(30, 370), 0)
+            
+    """
 
-    #This line of code is now shorten and done using sprite groups
-    #def draw(self, surface):
-        #surface.blit(self.image, self.rect)
+    def move(self):
+        self.rect.x += self.velocity[0]
+        self.rect.y += self.velocity[1]
 
+    def bounce(self):
+        self.velocity[0] = -self.velocity[0]
+        self.velocity[1] = random.randint(-8, 8)
 
-class Player(pygame.sprite.Sprite):
-    def __init__(self):
+    # This line of code is now shorten and done using sprite groups
+    # def draw(self, surface):
+    # surface.blit(self.image, self.rect)
+
+
+class Player_1(pygame.sprite.Sprite):
+    def __init__(self, coord):
         super().__init__()
         self.image = pygame.image.load("Player.png")
         self.rect = self.image.get_rect()
-        self.rect.center = (160, 520)
+        self.rect.center = coord
 
     def move(self):
         pressed_keys = pygame.key.get_pressed()
-        if pressed_keys[K_UP]:
+        if pressed_keys[K_UP] and self.rect.top > 0:
             self.rect.move_ip(0, -5)
-        if pressed_keys[K_DOWN]:
+        if pressed_keys[K_DOWN] and self.rect.bottom < 600:
```

Comment:
It would be better to go off your globals for the screen width and heights instead of hard coding 600. When you change the window size later you only have to change one place and won't have bugs in multiple places that also need updates.

This will let you go out of bounds if you are already right up to the boundary but not yet over it. 
This leads to another check, it would be good to "snap" to position if out of bounds. so if you are passed 600, at like 602, then force the position back to 600.

Reply:
Same applies to player 2 and "enemy"

Reply:
Ah yes, I agree!